impala/dbapi/hiveserver2.py

```diff
@@ -201,13 +201,49 @@ def _get_sleep_interval(self, start_time):
             return 0.5
         return 1.0
 
-    def executemany(self, operation, seq_of_parameters):
+    def _exec_single_query_insert(self, operation, seq_of_parameters):
```

Comment:
Instead of `_exec_single_query_insert()`, could you have a method called `_rewrite_inserts_single_query()` or something like that, which takes in `operation` and `seq_of_parameters` and returns the combined single SQL query string?